### PR TITLE
refactor(Image): mf-3376 introduce optimistic mode

### PR DESCRIPTION
**Pull request by UncleBill**
https://mask.atlassian.net/browse/MF-3376

image

### CODE REVIEW

File: packages/mask/src/manifest-v3.json [hunk 1]
```diff
@@ -21,6 +21,7 @@
                 "https://gateway.ipfscdn.io/ipfs/*",
                 "https://logo.nftscan.com/*",
                 "https://imagedelivery.net/*",
+                "https://assets.coingecko.com//*",
```
Comment (guanbinrui) [inline packages/mask/src/manifest-v3.json]:
```suggestion
                "https://assets.coingecko.com/*",
```

File: packages/plugins/RSS3/src/SNSAdaptor/components/FeedCard/TokenOperationCard.tsx [hunk 2]
```diff
@@ -125,7 +125,13 @@ export const TokenOperationCard: FC<TokenFeedCardProps> = ({ feed, ...rest }) =>
             </Typography>
             {metadata ? (
                 <div className={cx(classes.token, verbose ? classes.verboseToken : null)}>
-                    <Image classes={{ container: classes.tokenIcon }} src={metadata?.image} height={40} width={40} />
+                    <Image
+                        classes={{ container: classes.tokenIcon }}
+                        optimistic
+                        src={metadata?.image}
```
Comment (guanbinrui) [inline packages/plugins/RSS3/src/SNSAdaptor/components/FeedCard/TokenOperationCard.tsx:131]: fallback

**Review by guanbinrui — DISMISSED** [review]

File: packages/mask/src/manifest.json [hunk 3]
```diff
@@ -10,6 +10,6 @@
     "browser_action": { "default_popup": "popups.html" },
     "homepage_url": "https://mask.io",
     "description": "The portal to the new & open Internet. Send encrypted message and decentralized Apps right on top of social networks.",
-    "web_accessible_resources": ["js/*", "*.svg", "*.png", "*.css"],
+    "web_accessible_resources": ["js/*", "*.svg", "*.png", "*.css", "https://assets.coingecko.com/*"],
```
Comment (Jack-Works) [inline packages/mask/src/manifest.json:13]: I don't know it works! 🤔

File: packages/web3-shared/base/src/utils/resolver.ts [hunk 4]
```diff
@@ -443,6 +443,10 @@ export function resolveArweaveURL(url: string | undefined) {
     return urlcat('https://arweave.net/:str', { str: url })
 }
 
+/**
+ * Please do not use to resolve an image or an video resource, because that's
+ * not allowed by the cors agent server
```
Comment (Jack-Works) [inline packages/web3-shared/base/src/utils/resolver.ts:448]: can you explain more clearly?

File: packages/shared/src/UI/components/Image/index.tsx [hunk 5]
```diff
@@ -41,6 +41,8 @@ export interface ImageProps
     fallback?: URL | string | JSX.Element
     disableSpinner?: boolean
     containerProps?: HTMLProps<HTMLDivElement>
+    /** If optimistic is true, means that the resource is loadable */
+    optimistic?: boolean
```
Comment (Jack-Works) [inline packages/shared/src/UI/components/Image/index.tsx:50]: this is not a good name. maybe more clear, for example, `preferNative`

**Review by Jack-Works — CHANGES_REQUESTED** [review]

File: packages/web3-shared/base/src/utils/resolver.ts (hunk 4 above)
Comment (UncleBill) [reply packages/web3-shared/base/src/utils/resolver.ts:448]: Our CORS agent server just blocked image request few days ago.

File: packages/shared/src/UI/components/Image/index.tsx (hunk 5 above)
Comment (UncleBill) [reply packages/shared/src/UI/components/Image/index.tsx:50]: Like the Optimistic UI https://www.apollographql.com/docs/react/v2/performance/optimistic-ui/

File: packages/mask/src/manifest.json (hunk 3 above)
Comment (guanbinrui) [reply packages/mask/src/manifest.json:13]: If this declaration is reliable. Why not add more asset urls?

File: packages/web3-shared/base/src/utils/resolver.ts (hunk 4 above)
Comment (UncleBill) [reply packages/web3-shared/base/src/utils/resolver.ts:448]: Image proxy consume lots of traffic, costs too much money.

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**